jenkins/scripts/VersionSelectorScript.groovy

```diff
@@ -38,11 +45,32 @@ def buildExclusions = [
   [ /^cross-compiler-armv[67]-gcc-6/, anyType,     lt(12)  ],
 
   // Windows -----------------------------------------------
-  [ /vs2013/,                         anyType,     gte(6)  ],
-  [ /vs2015/,                         anyType,     lt(6)   ],
-  [ /vs2015/,                         anyType,     gte(10) ],
-  [ /vs2017/,                         anyType,     lt(10)  ],
-  [ /vs2019/,                         anyType,     lt(10)  ],
+  // https://github.com/nodejs/build/blob/master/doc/windows-visualstudio-supported-versions.md
+  // Release Builders - should only match one VS version per Node.js version
+  [ /vs2013/,                         releaseType, gte(6)        ],
+  [ /vs2015/,                         releaseType, ltGte(6, 10)  ],
+  [ /vs2017/,                         releaseType, ltGte(10, 14) ],
+  [ /vs2019/,                         releaseType, lt(14)        ],
+  // VS versions supported to compile Node.js - also matches labels used by test runners
+  [ /vs2013(-\w+)?$/,                 testType,    gte(6)        ],
+  [ /vs2015(-\w+)?$/,                 testType,    gte(10)       ],
+  [ /vcbt2015(-\w+)?$/,               testType,    gte(10)       ],
+  [ /vs2017(-\w+)?$/,                 testType,    lt(8)         ],
+  [ /vs2019(-\w+)?$/,                 testType,    lt(13)        ],
+  [ /vs2015-x86$/,                    testType,    gte(10)       ], // compile x86 only once
+  [ /vs2017-x86$/,                    testType,    ltGte(10, 14) ],
+  [ /vs2019-x86$/,                    testType,    lt(14)        ],
+  // VS versions supported to build add-ons
+  [ /vs2013-COMPILED_BY/,             testType,    gte(9)        ],
+  [ /vs2015-COMPILED_BY/,             testType,    noVer         ],
```

Comment:
👍 good idea to include these with `noVer` for completeness and documentation purposes, I like